src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -233,8 +236,10 @@ object GenomeReference {
   var defaultReference = GRCh37
 
   def addReference(gr: GenomeReference) {
-    if (references.contains(gr.name))
-      fatal(s"Cannot add reference genome. Reference genome `${ gr.name }' already exists.")
+    if (hasReference(gr.name))
+      fatal(s"Reference genome `${ gr.name }' already exists. Choose a reference name NOT in the following list:\n  " +
```

Comment:
I think you should make this symmetric with removeReferences. For example:

"Cannot remove reference genome. `${ gr.name }' already exists. Available references: `${ references.keys.mkString(", ") }'."

Even better, use your fancy list in both places instead of the mkString.